packages/core-components/src/layout/SignInPage/providers.tsx

```diff
@@ -202,3 +203,33 @@ export const useSignInProviders = (
 
   return [loading, elements];
 };
+
+/**
+ * A react hook to use to create login elements for the provided providers.
+ *
+ * @param providers - The providers to create login elements for.
+ * @param onSignInSuccess - The callback to call when a user has successfully signed in.
+ *
+ * @returns An array with a boolean indicating if the providers are still loading and the login elements.
+ *
+ * @public
+ * @example
+ * Example wrapping the providers with a grid container.
+ * ```tsx
+ * ...
+ * const providers = ['guest']
+ * const onSignInSuccess = (userIdentity: UserIdentity) => ...
+ *
+ * const [loading, providerElements] = useSignInProviderElements(
+ * providers,
+ *   onSignInSuccess,
+ * );
+ *
+ * return <Grid container>{providerElements}</Grid>;
```

Comment:
I'm a bit torn on this API, there's a lot baked into the hook, with the card + `Grid item` wrapping. Would a slightly higher level API work for you? I'm thinking a potentially `SignInProviderItems` that more or less mirrors the existing `SignInPage` but without the page wrapping, probably with the `Grid container` as the root element?